pkg/node/client.go

```diff
@@ -18,9 +19,19 @@ var NodeClientErr = errs.Class("node client error")
 
 // NewNodeClient instantiates a node client
 func NewNodeClient(self proto.Node) (Client, error) {
+	ca, err := provider.NewCA(context.Background(), 12, 4)
```

Comment:
you should make NewNodeClient take a ctx as an argument. context.Background() should basically never get called.